client/src/app/hooks/useAssessmentStatus.ts

```diff
@@ -0,0 +1,77 @@
+// hooks/useAssessmentStatus.js
+import { Assessment, Archetype, Application } from "@app/api/models";
+import { useFetchArchetypes } from "@app/queries/archetypes";
+import { useFetchAssessments } from "@app/queries/assessments";
+import { useMemo } from "react";
+
+export const useAssessmentStatus = (application: Application) => {
+  const { assessments } = useFetchAssessments();
+  const { archetypes } = useFetchArchetypes();
+
+  return useMemo(() => {
+    const applicationAssessments = assessments?.filter(
```

Comment:
`?.` suggest that `applicationAssessments` can be nulish later on - however we assume they are not.
Similar `inheritedArchetypes`